### ISSUE TITLE
Not using x-ms-code-generation-settings

### ISSUE THREAD

**Issue by iscai-msft** [Azure/autorest.modelerfour#184]
**Describe the issue**
When generating mgmt storage 2015-06-15, m4 is not accepting the [x-ms-code-generation-settings](https://github.com/Azure/azure-rest-api-specs/blob/825623d005ca24789e675db4fe75415c7585b7fe/specification/storage/resource-manager/Microsoft.Storage/stable/2015-06-15/storage.json#L7) override of [info][title]

** REQUIRED ** 
**ModelerFour** version: 4.7.207
**AutoRest-core** version: 3.0.6233
**CommandLine** : autorest ..\azure-rest-api-specs\specification\storage\resource-manager\readme.md --use=. --multiapi --python --python-mode=update --python-sdks-folder=..\azure-sdk-for-python\sdk\
OpenAPI/Swagger file: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2015-06-15/storage.json#L7

**Comment by fearthecowboy** [comment]
Yeah, we're not going to support `x-ms-code-generation-settings`  inside the swagger doc anymore. That's such a recipe for disaster, and at this point, it's time to migrate that stuff to a configuration file.

**Comment by lmazuel** [comment]
@fearthecowboy I see usage of this with the following options:
- name
- useDateTimeOffset
- internalConstructors

Could you confirm there is a way to move all three to Readme?
Also, could you fail parsing when you see this node, because right now we silently ignore that, and that is also a recipe to disaster :)

**Comment by fearthecowboy** [comment]
`name` is just `title: XXX`

Both `useDateTimeOffset` and `internalConstructors` should just be standard configuration settings. (ie, `--useDateTimeOffset` and  `--internalConstructors` ) 

I don't handle those two, but you should be able to get the settings the same as any other settings.

I can add a warning/error if `x-ms-code-generation-settings` is present

**Comment by lmazuel** [comment]
@fearthecowboy there is 193 Swagger out there with that, I think we should fail and not do a warning, this needs to be updated correctly.
I'll also ask a linter rule for that
